target/linux/ipq40xx/image/Makefile

```diff
@@ -131,6 +131,22 @@ define Device/glinet_gl-b1300
 endef
 TARGET_DEVICES += glinet_gl-b1300
 
+define Device/linksys_ea6350v3
+	$(call Device/FitImage)
+	DEVICE_TITLE := Linksys EA6350v3
+	DEVICE_DTS := qcom-ipq4018-ea6350v3
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	KERNEL_SIZE := 3145728
```

Comment:
Oh boy, 3145728 = 3 MiB is not going to last long. For 4.19 kernels (even with gzip -9!) I'm already at 3072373 bytes and that is a minimal build.

Could you please check if your uboot support lzma?

Reply:
Is defining the kernel size even needed since DTS defines a single UBI partition so buildroot can auto resize kernel and rootfs as needed?

Reply:
I would like to know what to do because I am not a Linux Kernel developer and I barely understand the DTS file.
As for the kenel size. I don't think it actually matters for the sysupgrade. But it does matter for a factory image flash, as the Linksys software does check the image.

Reply:
I will test but the Linksys Makefile defines the size. That is the reason of why we need to padd the kernel for getting a factory image.

And I really doubt that Linksys thought about a future-proof device.

Reply:
> But if they do check it then it would be best if you can test if U-boot supports LZMA by swapping $(call Device/FitImage) for $(call Device/FitImageLzma)

he's overwriting the KERNEL = production rule with his own gzip variant. He'll need to change that

Reply:
> I will test but the Linksys Makefile defines the size. That is the reason of why we need to padd the kernel for getting a factory image.

Question: Do you need the padding and fake rootfs only for the factory.bin? Because if so you should do 

IMAGE/factory.bin := append-kernel | append-uImage-fakehdr filesystem | pad-to $$$${KERNEL_SIZE} | append-ubi | linksys-image type=EA6350v3

and remove the special

KERNEL = ...

> And I really doubt that Linksys thought about a future-proof device.

Well, you could try to add

KERNEL_NAME := zImage

With this, the kernel will get an extra loader/decompressor that can do LZMA/xz.

Reply:
I removed the `KERNEL = ...` thing and used `$(call Device/FitImageLzma)`, but the device won't boot.
Using `$(call Device/FitImage)` and `KERNEL_NAME := zImage` does work.
AFAI can understand: zImage is a lzma compressed kernel with a self-extracting code. This 'kernel' looks like an 'uncompressed' Linux image for uboot. uboot will boot that but instad of running a kernel it will run the uncompressed decompressor. But `Device/FitImage` we compress it with gzip... isn't that counter-productive?
BTW the image I am getting is 8.5MiB uncompressed, 2.8MiB using gzip and 2.1MiB is the zImage.